### PR TITLE
Provide richer error messages for subnet CIDR validation

### LINKED ISSUE #1067: Make CIDR change message on subnet more useful
We should print the old & new CIDR values:

```
W1206 09:55:13.915271    9946 executor.go:84] error running task "subnet/blahblah": Field cannot be changed: CIDR
```

### PR DESCRIPTION
**Pull request by justinsb**
We just hit this in the real-world again, where a richer error message
would have been helpful.

Fix #1067

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2387)

### CODE REVIEW

File: upup/pkg/fi/utils/reflect.go [hunk 1]
```diff
@@ -211,3 +211,34 @@ func IsPrimitiveValue(v reflect.Value) bool {
 		return false
 	}
 }
+
+// FormatValue returns a string representing the value
+func FormatValue(value interface{}) string {
+	// Based on code in k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go
```
Comment (chrislovecnm) [inline upup/pkg/fi/utils/reflect.go]: Can we reuse the code????

Comment (chrislovecnm) [inline upup/pkg/fi/utils/reflect.go]: It is a nit pick ... your call.  Can we reuse the code rather than having code that is based on it.

**Comment by chrislovecnm** [comment]
Do we need this in 1.6?

File: upup/pkg/fi/utils/reflect.go (hunk 1 above)
Comment (justinsb) [inline upup/pkg/fi/utils/reflect.go]: It's buried in Error sadly.  We could reuse it, but it's probably cleaner to copy it.

**Comment by justinsb** [comment]
I don't see any harm in putting this into 1.6.

**Pull request merged**